prov/efa/src/efa_ep.c

```diff
@@ -699,22 +699,17 @@ int efa_ep_open(struct fid_domain *domain_fid, struct fi_info *user_info,
 		memcpy(ep->src_addr, user_info->src_addr, user_info->src_addrlen);
 	}
 
-	if (ep->domain->hmem_info[FI_HMEM_CUDA].initialized) {
-		/*
-		 * Set the default to required. NCCL plugin requires p2p, but
-		 * does not call setopt for this option in older NCCL plugin
-		 * versions.
-		 */
-		ep->hmem_p2p_opt = FI_HMEM_P2P_REQUIRED;
-	} else if (ep->domain->hmem_info[FI_HMEM_NEURON].initialized) {
-		/* Neuron requires p2p and supports no other modes. */
-		ep->hmem_p2p_opt = FI_HMEM_P2P_REQUIRED;
-	} else if (ep->domain->hmem_info[FI_HMEM_SYNAPSEAI].initialized) {
-		/* SynapseAI requires p2p and supports no other modes. */
-		ep->hmem_p2p_opt = FI_HMEM_P2P_REQUIRED;
-	} else {
-		/* no hmem devices, disable p2p */
-		ep->hmem_p2p_opt = FI_HMEM_P2P_DISABLED;
+	/* Set p2p opt to disabled by default */
+	ep->hmem_p2p_opt = FI_HMEM_P2P_DISABLED;
+
+	EFA_HMEM_IFACE_FOREACH_NON_SYSTEM(i) {
+		if (ep->domain->hmem_info[efa_hmem_ifaces[i]].initialized &&
+			ep->domain->hmem_info[efa_hmem_ifaces[i]].p2p_supported_by_device) {
+			ep->hmem_p2p_opt = ep->domain->hmem_info[efa_hmem_ifaces[i]].p2p_required_by_impl
```

Comment:
Not sure. If there are multiple interfaces, any one's  p2p_require_by_impl is true should make hmem_p2p_opt to P2P_REQUIRED.
Better wrap this up in a utility function

Reply:
> If there are multiple interfaces, any one's p2p_require_by_impl is true should make hmem_p2p_opt to P2P_REQUIRED. Better wrap this up in a utility function

That directly contradicts your concern from yesterday:

> It is possible for an instance to have both Neuron library and CUDA library installed. With this, neuron setting end up always overwrite CUDA setting.

The logic in this PR matches our current behavior. What you're suggesting would impose a tighter restriction on the p2p opt initialization. Maybe that's fine, but I want to make sure this is what we actually want.

Reply:
>Maybe that's fine, but I want to make sure this is what we actually want.

I agree to not change behavior in this PR.

Can you add a TODO note here about this? I think we need the tighter restrictions eventually.